frog/imports/ui/graph/DraggableAc.jsx

```diff
@@ -0,0 +1,215 @@
+    y: this.state.y
+  })
+
+  render() {
+    let {activity, editorMode} = this.props
+    let duration = convertTimeToPx(this.props.scale, activity.data.duration 
+                    ? activity.data.duration 
+                    : defaultTime)
+    return(
+      <div style = { { position: 'relative', zIndex: 0 } }>
+        <Rnd
+          ref = { c => { this.rnd = c } }
+          moveAxis={this.props.editorMode ? 'x' : 'none'}
+          id = {'drag_' + activity._id}
+          initial={{
+            x: this.defaultPosition().x,
+            y: this.defaultPosition().y,
+            height: boxHeight,
+            width: divStyle(duration).width
+          }}
+          isResizable= {{ top: false, right: editorMode, bottom: false, left: false, topRight: false,
+                          bottomRight: false, bottomLeft: false, topLeft: false }}
+          bounds={{left: textSizeAndMargin - rndMargin}}
+          onDrag={this.handleDrag}
+          onDragStop={this.handleStop}
+          onResize={this.handleResize}
+          onResizeStop={this.handleResizeStop}
+          moveGrid={[editorMode ? horizontalZoom : 0, 0]}
+          resizeGrid={[horizontalZoom, 0]}
+          zIndex={0}
+          style={divStyle(duration)}
+          >
+          <span style = { { position: 'relative', zIndex: 0 } }>
+            <div style = { { position: 'relative', zIndex: 0 } }>
+              <div style= { { position: 'absolute', zIndex: 0, left:-circleRadius } }>
+                <Anchor
+                onClick = { (event) => editorMode ? this.props.targetOperator(activity) 
+                                                : event.preventDefault() }
+                fill = 'white'
+                id = { 'target' + this.props.graphId + activity._id }
+                duration={duration}
+                />
+              </div>
+              <div style = { { position: 'absolute', zIndex: 0, right:-circleRadius } } >
+                <Anchor
+                onClick = { (event) => editorMode ? this.props.sourceOperator(activity) 
+                                                : event.preventDefault() }
+                fill = { this.props.isSourceClicked ? 'red' : 'white' }
+                id = { 'source' + this.props.graphId + activity._id}
+                duration={duration}
+                />
+              </div>
+            </div>
+            <div id = { activity._id } >
+              <a data-tip data-for = {'tip' + activity._id }>
+                <i className = 'fa fa-info' style = { { color: 'white' } }/>
+              </a>
+              {
+                <span>
+                  {this.props.editorMode && duration >= minRealBox ?
+                    <a
+                      onClick={(event) => this.props.delete(activity)}
+                      >
+                      <i className='fa fa-trash'  style= { { color: 'white' } }/>
+                    </a> : ''
+                  }
+                </span>
+              }
```

Comment:
Here you have:
```Javascript
 +              {
 +                <span>
 +                  {this.props.editorMode && duration >= minRealBox ?
 +                    <a
 +                      onClick={(event) => this.props.delete(activity)}
 +                      >
 +                      <i className='fa fa-trash'  style= { { color: 'white' } }/>
 +                    </a> : ''
 +                  }
 +                </span>
 +              }
```
Wouldn't it be the same to have:
```Javascript
 +                  {this.props.editorMode && duration >= minRealBox ?
 +                    <a onClick={(event) => this.props.delete(activity)}>
 +                      <i className='fa fa-trash'  style= { { color: 'white' } }/>
 +                    </a> : ''
 +                  }
```
?

I find a lot of useless brackets or divs in your code.